iaso/models/base.py

```diff
@@ -885,6 +888,37 @@ class Instance(models.Model):
     # See public_create_url workflow in enketo/README.md. used to tell we should export immediately
     to_export = models.BooleanField(default=False)
 
+    def __str__(self):
+        return "%s %s" % (self.form, self.name)
+
+    @property
+    def is_instance_of_reference_form(self) -> bool:
+        if not self.org_unit or not self.org_unit.org_unit_type:
+            return False
+        return self.org_unit.org_unit_type.reference_forms.filter(id=self.form_id).exists()
+
+    @property
+    def is_reference_instance(self) -> bool:
+        if not self.org_unit:
+            return False
+        return self.org_unit.reference_instances.filter(orgunitreferenceinstance__instance=self).exists()
+
+    def flag_reference_instance(self, org_unit: "OrgUnit") -> "OrgUnitReferenceInstance":
+        if not self.form:
+            raise ValidationError(_("The Instance must be linked to a Form."))
+        if not org_unit.org_unit_type:
+            raise ValidationError(_("The OrgUnit must be linked to a OrgUnitType."))
+        if not org_unit.org_unit_type.reference_forms.filter(id=self.form_id).exists():
```

Comment:
Could be replaced with the property:
```suggestion
        if not self.is_instance_of_reference_form():
```

Reply:
In the current state of affairs, it cannot because `org_unit` is an argument.

The reason why it's an argument are:

- the `Instance.org_unit` field is nullable (I don't konw if it's always set AND set to the right one?)
- the previous version of the code was acting like this